packages/ra-core/src/auth/usePermissionsOptimized.ts

```diff
@@ -13,7 +13,7 @@ const emptyParams = {};
 
 // keep a cache of already fetched permissions to initialize state for new
 // components and avoid a useless rerender if the permissions haven't changed
-const alreadyFetchedPermissions = { '{}': [] };
+const alreadyFetchedPermissions = { '{}': undefined };
```

Comment:
This was another regression introduced in 3.11. Before, WithPermissions injected `undefined` on mount. After, it injected an empty array.